administration/src/RegionProvider.tsx

```diff
@@ -1,12 +1,29 @@
-import {createContext, ReactNode, useState} from "react"
-import {Region} from "./generated/graphql";
+import {useQuery} from "@apollo/client";
+import {Button, Spinner} from "@blueprintjs/core";
+import {createContext, ReactNode, useContext} from "react"
+import {GetRegionsDocument, GetRegionsQuery, GetRegionsQueryVariables, Region} from "./generated/graphql";
+import {ProjectConfigContext} from "./project-configs/ProjectConfigContext";
+import {AuthContext} from "./AuthProvider";
 
-export const RegionContext = createContext<[Region | null, (region: Region | null) => void]>([null, () => {
-}])
+export const RegionContext = createContext<Region | null>(null)
 
 const RegionProvider = ({children}: { children: ReactNode }) => {
-    const [region, setRegionId] = useState<Region | null>(null)
-    return <RegionContext.Provider value={[region, setRegionId]}>
+    const {projectId} = useContext(ProjectConfigContext)
+    const [authContextData,] = useContext(AuthContext)
+    const userRegionId = authContextData?.administrator.regionId
+    const {loading, error, data, refetch} = useQuery<GetRegionsQuery, GetRegionsQueryVariables>(GetRegionsDocument, {
+        variables: {project: projectId}
+    })
+    if (loading) return <Spinner/>
+    if (error || !data) return <Button icon="repeat" onClick={() => refetch()}/>
+    const region = data.regions.find(region => region.id === userRegionId)
+    if (!region && typeof userRegionId === "number") {
+        return <>
+            <p>Your region was not found.</p>
+            <Button icon="repeat" onClick={() => refetch()}/>
+        </>
+    }
```

Comment:
I think a Provider should not be able to block the rendering of child components.

All of this could just be refactored into a hook: `useRegion` which calls `useQuery`. I would suspect `useQuery` to cache results to queries do not get re-executed.

Reply:
We might want to use some pattern for storing "global mutable/immutable state". There is `useReducer` now in react, which when paired with `useContext` makes redux unneccassary: https://reactjs.org/docs/hooks-reference.html#usereducer

In fact, `useState` is just a simpler interface to `useReducer`

Reply:
I could show you next week how I used both in some other project. We can also refactor this later though.
I'm not sure what is the "right" way of handling application state nowadays. btw, redux sucks.

Reply:
Hmm... blocking child components, because data they depend on hasn't finished loading yet, sounds reasonable to me. Not sure if other solutions would be a lot better. Is it just the name "Provider" you have problems with? Should we rename it to RegionsLoader or similar?
I don't think managing the state in a more complex way would help here :D

Reply:
Alright lets just merge this. In the end the code quality of the frontend does not matter right now much.

Basically the alternative is the following: Instead of blocking the rendering of the whole app at the Provider level, provide a null value and delay the rendering of each component which depends on the role.

Depending on the usecase either one or the other is better.

The name `Provider` is just fine here, as it is a React Provider.

Reply:
I'll just merge this for now. We can discuss some other approach later maybe. I think it is the easiest solution for now...